### ISSUE TITLE
Issues with jsx language loading after upgrade to prismjs 1.24.1 from 1.22.0

### ISSUE THREAD

**Issue by nage1234** [PrismJS/prism#3036]
**Information:**
- Prism version: 1.23.0 onwards
- Plugins: line-numbers
- Environment:  Node nodejs v11.0.0, Webpack 3.7.1, React 16.13.0 using latest version of babel-plugin-prismjs]

**Description**
in prismjs 1.22.0, this issue not reproducing. Once after upgrade to 1.23.0 or even the latest 1.24.1, i could see the issue, where jsx language is not loaded into Prism object.

Here are the snippets of the code.

Here is my babelrc file looks like:
{
  "presets": ["env", "react", "stage-1"],
  "env": {
    "development": {
      "presets": ["react-hmre"]
    },
    "lib": {
      "plugins": [
        "transform-react-constant-elements",
        "./babel-plugin-import-scss-to-css.js"
      ]
    },
    "production": {
      "plugins": ["transform-react-constant-elements"]
    }
  },
  "plugins": [
    [
      "prismjs",
      {
        "languages": ["javascript", "css", "markup", "jsx"],
        "plugins": ["line-numbers"]
      }
    ]
  ]
}

I haave a js file, where i am using the Prismjs and dynamically passing codes to this file to get it highlighted.

code-highlighter.js

**import Prism from 'prismjs/prism';
import Normalizer from 'prismjs/plugins/normalize-whitespace/prism-normalize-whitespace';
const codeNormalizer = new Normalizer({
  'remove-trailing': true,
  'remove-indent': true,
  'left-trim': true,
  'right-trim': true,
  'break-lines': 100,
  'remove-initial-line-feed': true,
  'tabs-to-spaces': 2,
  'spaces-to-tabs': 2,
  indent: 0
});
const simpleNormalizer = new Normalizer({
  'remove-trailing': true,
  'remove-indent': true,
  'left-trim': true,
  'right-trim': true,
  'remove-initial-line-feed': false
});
export const highlighter = (code, lang = 'jsx') =>
  Prism.highlight(codeNormalizer.normalize(code), Prism.languages[lang], lang);
export const normalize = (code, options) => simpleNormalizer.normalize(code, options);**

if u note the above code, i need to import Prism by changing the import from 'prismjs' to 'prismjs/prism'
otherwise, i am getting, "Prism is not defined" error.

if i peep into the Prism object in this code, i could not see jsx being loaded in the languages (Prism.languages object)

it seems, 1.23.0 or 1.24.1 (so far i have tested with these version of prismjs in my application) not backward compatible. i couldnt able to find in your docs, what exact changes went in from 1.22.0 to 123.0. Anything, which i need to change in my code, based on the changes went in for 1.23.0

i think still there is support for "jsx" language, but how come it is not showing up in the Prism.languages object.

I dont have working example, also, my project is confidential, i couldnt able to share it here. so, let me see whether i can do a small snippet of code changes to reproduce this one.

Any ide on this one could help.

i am not sure on Any dependency i miss for jsx language. it was working fine when i use prismjs 1.22.0

**Comment by RunDevelopment** [comment]
We haven't changed the dependencies of JSX since v1.22.0.

> if u note the above code, i need to import Prism by changing the import from 'prismjs' to 'prismjs/prism'

That's the problem. The Babel plugin searches for imports of [`prismjs` and nothing else](https://github.com/mAAdhaTTah/babel-plugin-prismjs/blob/587b00e25cd59dd18ed83adad4c5253fc0d57354/src/index.js#L9), so it won't detect `prismjs/prism`. `prismjs/prism` is a bundle that includes HTML, CSS, JS, and C-like. 

That's why JSX isn't loaded. You import a bundle that doesn't include JSX and (unintentionally) aren't using the babel plugin.

> otherwise, i am getting, "Prism is not defined" error.

That's strange. Does that also happen without the babel plugin?

Also, are you using the latest version of the babel plugin?

**Comment by nage1234** [comment]
Yes. i am using the latest version of babel plugin. for now, to make it work, as you suggested, i have imported jsx manually into the code-highlighting.js. now, it is working.

Regarding the other thing, still i am not sure, why this statement, "import Prism from 'prismjs'" not working. I didnt try without babel plugin. Let me try to get u an working example to reproduce this issue. i mean to say "reference error: Prism is not defined" issue

**Comment by RunDevelopment** [comment]
> Let me try to get u an working example

That would be great. Thank you!
